tests/app/views/contexts/summary/test_answer.py

```diff
@@ -37,18 +41,20 @@ def test_create_answer(return_to, return_to_block_id, is_in_repeating_section):
     assert answer.value == "An answer"
     assert answer.type == "date"
 
-    return_to_answer_id = answer.id
-    if not is_in_repeating_section:
-        return_to_answer_id += "-answer-item-id"
-
-    if return_to and return_to_block_id:
-        query_string = f"?return_to={return_to}&return_to_answer_id={return_to_answer_id}&return_to_block_id={return_to_block_id}"
-    elif return_to:
-        query_string = (
-            f"?return_to={return_to}&return_to_answer_id={return_to_answer_id}"
-        )
+    if return_to and return_to_block_id and return_to_answer_id:
```

Comment:
I find this test really hard to follow in it's current state, I know it had if statements previously but this should be avoided in tests. I would be in favour of moving the query string to the params e.g.
`@pytest.mark.parametrize(
    "return_to, return_to_block_id, is_in_repeating_section, return_to_answer_id, query_string",
    [
        (
            "section-summary",
            None,
            False,
            "answer-id",
            "?return_to=section-summary&return_to_answer_id=answer-id-answer-item-id,answer-id",
        ),
        (None, None, False, "answer-id", ""),
        (
            "calculated-summary",
            "total",
            False,
            "answer-id-answer-item-id",
            "?return_to=calculated-summary&return_to_answer_id=answer-id-answer-item-id,answer-id-answer-item-id&return_to_block_id=total",
        ),
        (
            "section-summary",
            None,
            True,
            "answer-id",
            "?return_to=section-summary&return_to_answer_id=answer-id,answer-id",
        ),
        (None, None, True, "answer-id-answer-item-id", ""),
        (
            "calculated-summary",
            "total",
            True,
            "answer-id",
            "?return_to=calculated-summary&return_to_answer_id=answer-id,answer-id&return_to_block_id=total",
        ),
        (
            "calculated-summary",
            "total",
            True,
            "calculated-summary-1",
            "?return_to=calculated-summary&return_to_answer_id=answer-id,calculated-summary-1&return_to_block_id=total",
        ),
    ],
)`

Reply:
Done, thanks.